test/unittest/unittest_nntrainer_tensor.cpp

```diff
@@ -1637,6 +1637,396 @@ TEST(nntrainer_Tensor, sum_02_p) {
 }
 
 TEST(nntrainer_Tensor, sum_03_p) {
+  int status = ML_ERROR_NONE;
+  const int batch = 3;
+  const int channel = 2;
+  const int height = 1;
+  const int width = 10;
+
+  nntrainer::Tensor input(batch, channel, height, width);
+  GEN_TEST_INPUT(input, i * (height * channel * width) + j * (height * width) +
+                          k * (width) + l + 1);
+  // Test for alpha == 1 and beta == 0 and dimension of reduced axis == 1
+  {
+    float ans_0_1_0[1][channel][height][width] = {
+      {{{63, 66, 69, 72, 75, 78, 81, 84, 87, 90}},
+       {{93, 96, 99, 102, 105, 108, 111, 114, 117, 120}}}};
+
+    float ans_1_1_0[batch][1][height][width] = {
+      {{{12, 14, 16, 18, 20, 22, 24, 26, 28, 30}}},
+      {{{52, 54, 56, 58, 60, 62, 64, 66, 68, 70}}},
+      {{{92, 94, 96, 98, 100, 102, 104, 106, 108, 110}}}};
+
+    float ans_2_1_0[batch][channel][1][width] = {
+      {{{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}},
+       {{11, 12, 13, 14, 15, 16, 17, 18, 19, 20}}},
+      {{{21, 22, 23, 24, 25, 26, 27, 28, 29, 30}},
+       {{31, 32, 33, 34, 35, 36, 37, 38, 39, 40}}},
+      {{{41, 42, 43, 44, 45, 46, 47, 48, 49, 50}},
+       {{51, 52, 53, 54, 55, 56, 57, 58, 59, 60}}}};
+
+    float ans_3_1_0[batch][channel][height][1] = {
+      {{{55}}, {{155}}}, {{{255}}, {{355}}}, {{{455}}, {{555}}}};
+
+    nntrainer::Tensor result_0_1_0 = input.sum(0, 1);
+    nntrainer::Tensor result_1_1_0 = input.sum(1, 1);
+    nntrainer::Tensor result_2_1_0 = input.sum(2, 1);
+    nntrainer::Tensor result_3_1_0 = input.sum(3, 1);
+
+    for (unsigned int i = 0; i < result_0_1_0.batch(); ++i) {
+      for (unsigned int l = 0; l < result_0_1_0.channel(); ++l) {
+        for (unsigned int j = 0; j < result_0_1_0.height(); ++j) {
+          for (unsigned int k = 0; k < result_0_1_0.width(); ++k) {
+            if (ans_0_1_0[i][l][j][k] != result_0_1_0.getValue(i, l, j, k)) {
```

Comment:
recommend to wrap `ans_*_1_0` with a tensor and directly compare tensors.
you can use a 4D vector of floats instead of 4D float array or wrap array with vector or add constructor in Tensor for 4D array.

Reply:
Done